### DISCUSSION TITLE
Deleting tags.data and undo.data

### DISCUSSION THREAD

**Discussion by iameru** [GothenburgBitFactory/timewarrior#593] [Q&A] [answered]
I was using `timew` on two different devices for the past months, while one of them has a Database reaching back years. I want to merge them and just sync the directories with the data in the future as suggested ( #428 #340 #359 )

I found a `tags.data` in both sources, of course they don't match. I don't really find info on the importance of the file, however in the source code it _looks like to me_  as if it could just be deleted and then it would be regenerated. I didn't look too much into it though.

So can I just delete `tags.data` in my merging? the times are easier to merge.

Can I also just delete `undo.data` if I am content with the state of the database as it is?

**Comment by lauft** [comment, marked as answer]
As you found out, `tags.data` is regenerated from the database files if missing. (It contains a list of your tags and their respective counts.) So deleting it should not be a problem.

When you are content with the state of your DB, you do not require a `undo`. Deleting `undo.data` will only prevent you from undoing your previous commands. As those might not apply to the state of the DB after the merge, there is also a chance undoing them won't work anymore. By default, the `undo.data` grows indefinitely. You can limit the size of the file by setting the configuration `journal.size` to the number of lines you want to keep (`-1` meaning `infinite`, `0` no journaling at all).

In any way it is advisable to do a backup of your files before the merge. 😇

**Reply by iameru** [reply]
Thanks. I was able to easily merge the timedelta files and everything went smooth, regenerating a proper `tags.data`.
